javaparser-core/src/main/java/com/github/javaparser/ast/validator/chunks/ModifierValidator.java

```diff
@@ -49,7 +49,7 @@ public void visit(ClassOrInterfaceDeclaration n, ProblemReporter reporter) {
 
     private void validateClassModifiers(ClassOrInterfaceDeclaration n, ProblemReporter reporter) {
         if (n.isTopLevelType()) {
-            validateModifiers(n, reporter, PUBLIC, ABSTRACT, FINAL, STRICTFP);
+            validateModifiers(n, reporter, PUBLIC, ABSTRACT, STATIC, FINAL, STRICTFP);
```

Comment:
It is definitely not possible in java.jj. But it is possible to add a code that will go through the StubUnit after it is parsed and look for classes that contain "$" in names and try put it inside another if found the match for outer class. But this will happen only after the stub file is already parsed into the StubUnit.
Should I work on it?